src/librustc_driver/lib.rs

```diff
@@ -666,6 +666,9 @@ impl RustcDefaultCalls {
                             // MetaItemKind::List items in the configuration returned by
                             // `build_configuration`.
                             panic!("MetaItemKind::List encountered in default cfg")
+                        } else {
+                            // There also shouldn't be literals.
+                            panic!("Unexpected MetaItem literal in default cfg!")
```

Comment:
Change both? It would be nice to give more descriptive errors with some of the values on-hand, too (eg emitting the `config` name that caused the problem).